### ISSUE TITLE
Can't install codeworld-api with cabal due to version bound

### ISSUE THREAD

**Issue by ThePhoenixCoding** [google/codeworld#1497]
**Description**

Similar to [this issue here](https://github.com/google/codeworld/issues/1488) I can't install the codeworld-api-0.7 due to the version bound for ghc-prim being < 0.7 but version after fresh install being = 0.7. Error code from Git Bash:

alexa@Phoenixbook MINGW64 ~
$ cabal install --lib QuickCheck HUnit codeworld-api-0.7.0
Resolving dependencies...
cabal.exe: Could not resolve dependencies:
[__0] trying: HUnit-1.6.2.0 (user goal)
[__1] trying: base-4.15.0.0/installed-4.15.0.0 (dependency of HUnit)
[__2] next goal: codeworld-api (user goal)
[__2] rejecting: codeworld-api-0.7.0 (conflict: base =>
ghc-prim==0.7.0/installed-0.7.0, codeworld-api => ghc-prim>=0.3.1 && <0.7)
[__2] rejecting: codeworld-api-0.6.0, codeworld-api-0.5.0,
codeworld-api-0.4.0, codeworld-api-0.3.1, codeworld-api-0.3,
codeworld-api-0.2.5, codeworld-api-0.2.4, codeworld-api-0.2.3,
codeworld-api-0.2.2.1, codeworld-api-0.2.2.0, codeworld-api-0.2.1.0,
codeworld-api-0.2.0.0, codeworld-api-0.1.0.1 (constraint from user target
requires ==0.7.0)
[__2] fail (backjumping, conflict set: base, codeworld-api)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, codeworld-api, HUnit

**Comment by cdsmith** [comment]
Bound is updated, and Hackage revision has been published.

**Comment by marcellussiegburg** [comment]
A new culprit has been found
(see https://matrix.hackage.haskell.org/#/package/codeworld-api/0.7.0/ghc-8.10.1@1617628302, i.e.
```
1617680396.225 Resolving dependencies...
1617680396.953 cabal: Could not resolve dependencies:
               [__0] trying: codeworld-api-0.7.0 (user goal)
               [__1] next goal: template-haskell (dependency of codeworld-api)
               [__1] rejecting: template-haskell-2.16.0.0/installed-2.16.0.0 (conflict: codeworld-api => template-haskell>=2.8 && <2.16)
               [__1] skipping: template-haskell-2.17.0.0, template-haskell-2.16.0.0 (has the same characteristics that caused the previous version to fail: excluded by constraint '>=2.8 && <2.16' from 'codeworld-api')
               [__1] rejecting: template-haskell-2.15.0.0, template-haskell-2.14.0.0, template-haskell-2.13.0.0, template-haskell-2.12.0.0, template-haskell-2.11.1.0, template-haskell-2.11.0.0, template-haskell-2.10.0.0, template-haskell-2.9.0.0, template-haskell-2.8.0.0, template-haskell-2.7.0.0, template-haskell-2.6.0.0, template-haskell-2.5.0.0, template-haskell-2.4.0.1, template-haskell-2.4.0.0, template-haskell-2.3.0.1, template-haskell-2.3.0.0, template-haskell-2.2.0.0 (constraint from project config TODO requires installed instance)
               [__1] fail (backjumping, conflict set: codeworld-api, template-haskell)
               After searching the rest of the dependency tree exhaustively, these were the goals I've had most trouble fulfilling: template-haskell, codeworld-api
```
)

**Comment by cdsmith** [comment]
Which GHC version is this?

**Comment by marcellussiegburg** [comment]
This report here is for `ghc-8.10.1`.

**Comment by cdsmith** [comment]
I've updated this bound.  But I cannot really test effectively without knowing how to do so.  Can you share the GHC version and cabal file (if any) you are using to build?

**Comment by cdsmith** [comment]
I've relaxed that bound, but also am verifying that I can install with GHC 8.10

**Comment by cdsmith** [comment]
Still doesn't build now because codeworld-api depends on an old reflex that doesn't build with base 0.14.  So more changes are needed

**Comment by cdsmith** [comment]
It looks like there's no reflex support for GHC 8.10 and above.  That's too bad, and it blocks building codeworld-api, which depends on reflex.

**Comment by marcellussiegburg** [comment]
It seems we have to wait for this issue to be closed reflex-frp/reflex#443.
